### PR TITLE
Add interpolation feature

### PR DESCRIPTION
**Pull request by frmadem**
Related to #12

### CODE REVIEW

**Review by arvegadev — APPROVED** [review]
LGTM

File: utils/context_resolutor.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    if(!resolutor){
+
+      resolved = __getResolutor(level)
+    
+    }
+    else{
+    
+      if(!resolved[l]){
+
+        throw `COULD NOT RESOLVE ${key}`
```
Comment (alambike) [inline utils/context_resolutor.js:30]:
```suggestion
        throw `COULD NOT RESOLVE ${{key}}`
```

File: utils/context_resolutor.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+  function __getResolutor(level){
+
+    switch(level){
+
+      case "secrets":
+        return github.context.secrets
+      case "env"
+        return github.context.env
+      default:
```
Comment (alambike) [inline utils/context_resolutor.js:52]: Are you sure that the secrets and env contexts are available in this way?
https://github.com/actions/toolkit/blob/main/packages/github/src/context.ts

File: fixtures/build5.test.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  triggers:
+    push:
+      branches:
+        - dev
+        - main
+  build_args: 
+    APP: foo
+    App2: lol
+    PAT: ${{secret.PAT}}
+  dockerfile: Dockerfile.outro
```
Comment (alambike) [inline fixtures/build5.test.yaml:11]: Missing test for env context

File: utils/context_resolutor.js (hunk 1 above)
Comment (frmadem) [reply utils/context_resolutor.js:30]: No @alambike this is a [template string](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals).

File: utils/context_resolutor.js (hunk 2 above)
Comment (frmadem) [reply utils/context_resolutor.js:52]: Not sure at all, I was going to test it in this way an make further corrections.

Comment (frmadem) [reply utils/context_resolutor.js]: Fixed for the env object.

Comment (alambike) [reply utils/context_resolutor.js]: `${{secrets}}` cannot be interpolated directly in an action execution, they must be passed from the workflow, as env or input variable. Update related issue #12
https://docs.github.com/en/actions/security-guides/encrypted-secrets#accessing-your-secrets
image

File: fixtures/build5.test.yaml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+my-flavour:
+  triggers:
+    push:
+      branches:
+        - dev
+        - main
+  build_args: 
+    APP: foo
+    App2: lol
+    PAT: ${{secret.PAT}}
```
Comment (alambike) [inline fixtures/build5.test.yaml:10]:
```suggestion
    PAT: ${{env.PAT}}
```

File: test/build.test.js [hunk 5]
```diff
@@ -101,3 +101,39 @@ test('Build get flavours by pre-release trigger', () => {
...
 })
+
+test('Build performs values interpolation', () => {
+
+  const build_data = fs.readFileSync("./fixtures/build5.test.yaml")
+
+  const build = new Build(build_data, function(value){
+
+    // mock of context resolutor
+    if(value === "${{secret.PAT}}"){
```
Comment (alambike) [inline test/build.test.js:112]:
```suggestion
    if(value === "${{env.PAT}}"){
```

File: fixtures/build5.test.yaml [hunk 6]
```diff
@@ -0,0 +1,26 @@
+my-flavour:
+  triggers:
+    push:
+      branches:
+        - dev
+        - main
+  build_args: 
+    APP: foo
+    App2: lol
+    PAT: ${{env.PAT}}
```
Comment (alambike) [inline fixtures/build5.test.yaml:10]: I would test the following expressions too:

```yaml
PAT: ${{ env.PAT }}
PAT: "${{env.PAT}}"
PAT: '${{env.PAT}}'
PAT: "${{ env.PAT }}"
...
```

**Review by alambike — CHANGES_REQUESTED** [review]

File: fixtures/build5.test.yaml (hunk 6 above)
Comment (frmadem) [reply fixtures/build5.test.yaml:10]: solved and tested.

Comment (alambike) [reply fixtures/build5.test.yaml:10]: I miss a test with single quotes

**Review by alambike — APPROVED** [review]

**Pull request merged**